experiment/measurer/measure_manager.py

```diff
@@ -88,33 +83,77 @@ def get_job_timeout():
+    that they can be measured."""
+    experiment = experiment_config['experiment']
+    max_total_time = experiment_config['max_total_time']
     max_cycle = _time_to_cycle(max_total_time)
+    logger.info('Enqueuing measure jobs.')
     unmeasured_snapshots = get_unmeasured_snapshots(experiment, max_cycle)
+    jobs = enqueue_measure_jobs(unmeasured_snapshots, queue)
+    logger.info('Done enqueuing jobs.')
+    return jobs
 
-    if not unmeasured_snapshots:
-        return False
 
+def enqueue_measure_jobs(unmeasured_snapshots, queue):
+    """Add jobs to measure each snapshot in |unmeasured_snapshots| to |queue|
+    and returns them."""
     job_timeout = get_job_timeout()
-    results = [
+    jobs = [
         queue.enqueue(measure_worker.measure_trial_coverage,
                       unmeasured_snapshot,
-                      job_timeout=job_timeout)
+                      job_timeout=job_timeout,
+                      result_ttl=job_timeout,
+                      ttl=job_timeout)
         for unmeasured_snapshot in unmeasured_snapshots
     ]
+    return jobs
 
-    # Poll the queue for snapshots and save them in batches until the pool is
-    # done processing each unmeasured snapshot. Then save any remaining
-    # snapshots.
+
+def enqueue_next_measure_job(unmeasured_snapshot, experiment_config, queue):
+    """Adds a job to measure the next snapshot after |unmeasured_snapshot| to
+    |queue| and returns the job."""
+    if _time_to_cycle(
+            experiment_config['max_total_time']) == unmeasured_snapshot.cycle:
+        return None
+    next_unmeasured_snapshot = measure_worker.SnapshotMeasureRequest(
+        unmeasured_snapshot.fuzzer, unmeasured_snapshot.benchmark,
+        unmeasured_snapshot.trial_id, unmeasured_snapshot.cycle + 1)
+    job = enqueue_measure_jobs([next_unmeasured_snapshot], queue)[0]
+    return job
+
+
+def ready_to_measure():
+    """Returns True if we are ready to start measuring."""
+    experiment_folders_dir = get_experiment_folders_dir()
+    return exists_in_experiment_filestore(experiment_folders_dir)
+
+
+def measure_all_trials(experiment_config: dict, queue) -> bool:  # pylint: disable=too-many-branches,too-many-statements
+    """Get coverage data (with coverage runs) for all active trials. Note that
+    this should not be called unless multiprocessing.set_start_method('spawn')
+    was called first. Otherwise it will use fork which breaks logging."""
+    logger.info('Measuring all trials.')
+
+    if not ready_to_measure():
+        return True
+
+    jobs = enqueue_measure_jobs_for_unmeasured(experiment_config, queue)
+    if not jobs:
+        # If we didn't enqueue any jobs, then there is nothing to measure.
+        return False
+
+    initial_jobs = {job.id: job for job in jobs}
+
+    # TODO(metzman): Move this to scheduler or it's own process. It's here for
+    # now because the scheduler quits too early.
```

Comment:
Consider removing this TODO, as we might be able to skip these steps, and get rid of the scheduler completely (as in queue arch proposal)

Reply:
I need to comment on that. A lot of the scheduler now involves controlling resource consumption, I assume it will be more of a refactor than a "remove" (much like how measurer wasn't removed but was split in two), but maybe you have something else in mind then me.

Reply:
I think there isn't harm in leaving the todo for now.